src/main.rs

```diff
@@ -62,10 +62,11 @@ fn run_target_program(elf_path: &Path, chip_name: &str, opts: &cli::Opts) -> any
     let probe = probe::open(opts)?;
 
     let probe_target = target_info.probe_target.clone();
+    let permissions = Permissions::default().allow_erase_all();
```

Comment:
Could you avoid obtaining this permission if you are not going to erase all?

Reply:
Yes, makes sense!